### ISSUE TITLE
When I send a POST request to "/api/request" with files:

### ISSUE THREAD

**Issue by bystro** [ubirak/rest-api-behat-extension#87]
It seems to be useful to test requests with files included.

I use https://github.com/danielm/uploader in my projects.
Dmuploader posts files to my REST API ex. /api/order/attachment-file
API responses success or false information.

Very precursory scenario format ...
```
Scenario: Privileged user attaches the file to the order 
When I send a POST request to "/api/order/attachment-file" with files:
"""
        {
            "files" : [{"fixtures/test-file1.png", "fixtures/test-file2.png"}],
        }
 """
Then the response status code should be 200
And the response should be in JSON
And the JSON should be equal to:
      """
        {
          "success": true,
          "data": 
              {
                  "success_message": "Files attached succesfully"
              }
      }
      """
```

What do you think?

**Comment by tyx** [comment]
Why not !

But I just guess we should use table notation instead of string notation, like
```
When I send a POST request to "/api/order/attachment-file" with files attached:
    | files                   |
    | fixtures/test-file1.png |
    | fixtures/test-file2.png |
```

It would be more simple to use and to code !

**Comment by bystro** [comment]
I wonder, how to make a mixture with posting files and body using existing "I send a POST request to "echo" with body:". ?

```
Wnen I send a POST request to "echo" with body:
"""
        {
            "username" : "pablo",
            "password": "money"
        }
        """
     with files attached:
    | files                   |
    | fixtures/test-file1.png |
    | fixtures/test-file2.png |
```

**Comment by stephpy** [comment]
Not sure if Behat syntax accepts it. Did you test it ?

**Comment by bystro** [comment]
I haven't test it yet.
I'm looking for notation with only one "send request to".

It would be very ackward if I would sent it as following...
```
Wnen I send a POST request to "echo" with body:
"""
 {
     "username" : "pablo",
     "password": "money"
 }
"""
And When I send a POST request to "echo" with files attached:
 | files                   |
 | fixtures/test-file1.png |
 | fixtures/test-file2.png |
```

**Comment by stephpy** [comment]
We can consider this:

```
Wnen I send a POST request to "echo" with:
    | body                   | {.........}
    | files[0]                | ....
    | files[1]                | ....
```

**Comment by bystro** [comment]
Have anyone checked if and how it is implemented at other extensions?

**Comment by tyx** [comment]
The only option we get I guess is to have 2 steps with a first one that push state (like [headers](https://github.com/ubirak/rest-api-behat-extension/blob/master/src/RestApiContext.php#L76))

**Comment by bystro** [comment]
@tyx could you write an example?

**Comment by tyx** [comment]
```
Given I attach files:
 | files                   |
 | fixtures/test-file1.png |
 | fixtures/test-file2.png |
When I send a POST request to "echo":
"""
 {
     "username" : "pablo",
     "password": "money"
 }
"""
```

**Comment by bystro** [comment]
I'm working on the issue today.

The problem is to send payload with json and files(multipart/form-data) together.
When I'm sending payload using "Content-Type: multipart/form-data; boundary=..." header, the json payload does not exists in a request body - "Wnen I send a POST request to "echo" with body:".

**Comment by bystro** [comment]
https://stackoverflow.com/questions/48009305/how-to-post-http-request-with-json-and-files-payload

**Comment by tyx** [comment]
indeed ! And it makes sense as we should use a different content type.

I just had a look and in one app we used base_64 workaround.

**Comment by bystro** [comment]
I think that If files are included, json attributes should be placed in multipart/form-data body.

But a big quession is if most restful frameworks will handle this like ZendFramework does
https://github.com/zendframework/zend-mvc/blob/master/src/Controller/AbstractRestfulController.php#L471

**Comment by tyx** [comment]
But how could you guess you add json in body without correct content type?

**Comment by tyx** [comment]
Fixed by #94
